pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -409,13 +418,22 @@ func (h *handler) OnRemove(key string, obj runtime.Object) (runtime.Object, erro
 }
 
 func (h *handler) doRemove(infra *infraObject) (runtime.Object, error) {
-	obj, err := h.run(infra, false)
+	state, _, err := h.run(infra, false)
```

Comment:
for other reviewers, the patch/update below was moved here from the run function